embedded/sql/stmt.go

```diff
@@ -93,6 +94,8 @@ const (
 type SQLStmt interface {
 	isDDL() bool
 	CompileUsing(e *Engine, implicitDB *Database, params map[string]interface{}) (ces, des []*store.KV, db *Database, err error)
+	InferParameters(e *Engine) (params map[string]SQLValueType, err error)
```

Comment:
I think we should provide InferParameter from database package.
Using engine is not pratical since server is using database package to expose api.

Reply:
this is the internal implementation, but the upper layers will expose it as you describe